rust/tools/authorization-logic/src/souffle/universe_translation.rs

```diff
@@ -174,6 +187,25 @@ fn populate_constant_type_environment_prog(
     constant_type_environment
 }
 
+fn populate_constant_type_environment_rvalue(
+    relation_type_environment: &RelationTypeEnv,
+    constant_type_environment: &mut HashMap<String, AstType>,
```

Comment:
It would be more functional to take the map as a value and return the populated map rather than taking a mutable reference and returning void if that's possible.

Reply:
I think making this overall collection of functions work that way would entail making `add_typing` do the same, which would entail making a copy of the entire hashmap inside `add_typing` because the standard libarary calls for updating hashmaps work by mutation rather than by returning a value and passing ownership to the return.